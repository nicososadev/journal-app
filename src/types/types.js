export const types = {

    login: '[Auth] Login',
    logout: '[Auth] Logout',

    uiSetError: '[UI] Set Error',
    uiRemoveError: '[UI] Remove Error',

    uiStartLoading: '[UI] Start loading',
    uiFinishLoading: '[UI] Finish loading',

    notesCreateNote: '[Notes] Create Note',
    notesUpdateNote: '[Notes] Update Note',
    notesDeleteNote: '[Notes] Delete Note',
    notesActive: '[Notes] Set Active Note',
    notesLoad: '[Notes] Load Notes',
    notesImgUrl: '[Notes] Note Image Url',
    notesLogoutCleaning: '[Notes] Logout Cleaning',

}