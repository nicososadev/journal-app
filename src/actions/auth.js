import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { types } from '../types/types'

export const loginManager = (email, password) => {

    return (dispatch) => {

        setTimeout(() => {
            
            dispatch( login(123, 'Juan') )
        }, 3500);
    }
}

export const googleLoginManager = () => {
    
    return (dispatch) => {

        firebase.auth().signInWithPopup(googleAuthProvider)
            .then( ({ user }) => {
                dispatch(
                    login(user.uid, user.displayName)
                )
            })
    }
}

export const login = (uid, displayName) => {

    return {
        type: types.login,
        payload: {
            uid,
            displayName
        }
    }
}