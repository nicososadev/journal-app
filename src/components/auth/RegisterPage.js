import React, { Fragment } from 'react'
import { Link } from 'react-router-dom'

export const RegisterPage = () => {
    return (
        <Fragment>
            <h3 className="auth__title">Register</h3>

            <form>
                <input className="auth__input" type="text" name="name" placeholder="Name" />
                <input className="auth__input" type="text" name="email" placeholder="Email" />
                <input className="auth__input" type="password" name="password" placeholder="Password" />
                <input className="auth__input" type="password2" name="password2" placeholder="Confirm Password" />

                <button
                    className="btn btn-primary btn-block mt-1"
                    type="submit"
                >
                    Register
                </button>

                <Link className="link mt-1" to="/auth/login">Already have an Account</Link>
        

            </form>
        </Fragment>
    )
}
