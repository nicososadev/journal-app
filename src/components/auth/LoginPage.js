import React, { Fragment } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import { googleLoginManager, loginManager } from '../../actions/auth'
import { useForm } from '../../hooks/useForm'

export const LoginPage = () => {

    const dispatch = useDispatch()

    const { loading } = useSelector(state => state.ui)

    const [formValues, handleInputChange] = useForm({
        email: 'nicososa@gmail.com',
        password: '123456'
    })

    const { email, password } = formValues

    const handleLogin = (e) =>{
        e.preventDefault()
        
        dispatch( loginManager(email, password) )
    }

    const googleLogin = () => {
        
        dispatch( googleLoginManager() )
    }



    return (

        <Fragment>
            <h3 className="auth__title">Login</h3>

            <form className="animate__animated animate__fadeIn animate__faster" onSubmit={ handleLogin } >
                <input className="auth__input" type="text" name="email" value={email} onChange={handleInputChange} placeholder="Email" />
                <input className="auth__input" type="password" name="password" value={password} onChange={handleInputChange} placeholder="Password" />

                <button
                    className="btn btn-primary btn-block"
                    type="submit"
                    disabled={loading}
                >
                    Login
                </button>

                <div className="auth__social-networks">
                    <p>Login with Social Network</p>
                    <div onClick={googleLogin} className="google-btn">
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with google</b>
                        </p>
                    </div>
                    <Link className="link mt-1" to="/auth/register">Create new Account</Link>
                </div>

            </form>
        </Fragment>
    )
}
