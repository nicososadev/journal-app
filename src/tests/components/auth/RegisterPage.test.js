import '@testing-library/jest-dom'
import React from 'react'
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import { RegisterPage } from '../../../components/auth/RegisterPage'
import { MemoryRouter } from 'react-router-dom'
import { types } from '../../../types/types'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        notes: [],
        active: null
    }
    
}

let store = mockStore(initialState)

describe('Pruebas en el componente RegisterPage', () => {
    
    const wrapper = mount(
        <MemoryRouter>
            <Provider store={store}>
                <RegisterPage/>
            </Provider>
        </MemoryRouter>
    )

    test('Debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe hacer el dispach de la accion respectiva', () => {

        const emailField = wrapper.find('input[name="email"]')

        emailField.simulate('change', {
            target: {
                value: '',
                name: 'email'
            }
        })

        wrapper.find('form').simulate('submit')

        const actions = store.getActions()

        expect( actions[0] ).toEqual({
            type: types.uiSetError,
            payload: 'Email is not correct'
        })
        
    })

    test('Debe mostrar correctamente la caja de alerta con el mensaje de error', () => {
        
        const initialState = {
            auth: {},
            ui: {
                loading: false,
                msgError: 'Email is not correct'
            },
            notes: {
                notes: [],
                active: null
            }
            
        }
        
        let store = mockStore(initialState)
            
        const wrapper = mount(
            <MemoryRouter>
                <Provider store={store}>
                    <RegisterPage/>
                </Provider>
            </MemoryRouter>
        )

        const alertBox = wrapper.find('.auth__alert-error')

        expect( alertBox.exists() ).toBe( true )
        expect( alertBox.text().trim() ).toBe( initialState.ui.msgError )

    })
    
    
    
})
