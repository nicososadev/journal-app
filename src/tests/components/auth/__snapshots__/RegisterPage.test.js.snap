// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente RegisterPage Debe mostrarse correctamente 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Nicolas"
    />
    <input
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="nicososa@gmail.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password2"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block mt-1"
      type="submit"
    >
      Register
    </button>
    <a
      className="link mt-1"
      href="/auth/login"
      onClick={[Function]}
    >
      Already have an Account
    </a>
  </form>,
]
`;
